src/rt/critical_.d

```diff
@@ -25,7 +25,8 @@ extern (C) void _d_critical_init()
 
 extern (C) void _d_critical_term()
 {
-    for (auto p = head; p; p = p.next)
+    auto h = cast()head;
```

Comment:
not so sure about just casting shared away, can you add a comment which explains why you think that this pointer is only accessed by a single thread at a time?

Reply:
This runs as the runtime is being deinitialised, access is guaranteed to be single threaded.

Reply:
please add a comment :)

Reply:
Done,